### PR TITLE
feat: regenerate schemas

### PR DESCRIPTION
**Pull request by Tomas2D**
- Regenerate schemas (Text Generation, Moderation, Experimental)
- Schema renaming / deprecation
- Extend our schema generator - custom fields/methods / parameters
- Preserve the order of generated schema models
- Handle newer LangChain version
- Ruff update

### CODE REVIEW

File: src/genai/extensions/localserver/schema.py [hunk 1]
```diff
@@ -10,13 +12,18 @@ class TextGenerationCreateRequest(_TextGenerationCreateRequest):
...
     """Following properties are required for local server"""
 
-    model_id: str
+    model_id: str = Field(...)
     parameters: TextTokenizationParameters = Field(default_factory=lambda: TextTokenizationParameters())
+    input: Optional[Union[str, list[str]]] = None
+    prompt_id: Optional[str] = None
 
 
 __all__ = ["TextGenerationCreateRequest", "TextTokenizationCreateRequest"]
```
Comment (Tomas2D) [inline src/genai/extensions/localserver/schema.py:29]: Even though inheritance is used, I had to import all the model's properties. When you do something like this.

```python
class TextTokenizationCreateRequest(_TextTokenizationCreateRequest):
    model_id: str = Field(...)
```

You won't be able to initiate it (Pydanti unknown error)—feel free to find out why.

File: scripts/types_generator/schema_aliases.yaml [hunk 2]
```diff
@@ -89,86 +87,168 @@ alias:
...
-    - _TextChatStreamCreate_moderation_hap_tokens
-    - _TextChatStreamCreate_moderation_stigma_tokens
-    - _TextChatStreamCreate_moderation_implicit_hate_tokens
-    - _TextGenerationStreamCreate_moderation_hap_tokens
-    - _TextGenerationStreamCreate_moderation_stigma_tokens
-    - _TextGenerationStreamCreate_moderation_implicit_hate_tokens
     - _TextModerationCreate_results_hap_tokens
-    - _TextModerationCreate_results_stigma_tokens
-    - _TextModerationCreate_results_implicit_hate_tokens
+    - _TextModerationCreate_results_social_bias_tokens
```
Comment (jezekra1) [inline scripts/types_generator/schema_aliases.yaml:153]: Uff 😅  That's a lot of aliases

File: scripts/types_generator/schema_aliases.yaml [hunk 3]
```diff
@@ -609,3 +691,98 @@ model_extensions:
...
+      - name: social_bias
+        body: return ModerationSocialBias(input=ModerationSocialBiasInput(enabled=value), output=ModerationSocialBiasOutput(enabled=value)) if isinstance(value, bool) else value
+
+  ModerationHAP:
+    validators:
+      - name: input
+        body: return ModerationHAPInput(enabled=value) if isinstance(value, bool) else value
+      - name: output
+        body: return ModerationHAPOutput(enabled=value) if isinstance(value, bool) else value
+    model_post_init: |
```
Comment (jezekra1) [inline scripts/types_generator/schema_aliases.yaml:715]: 👌

File: scripts/types_generator/assets/codegen/templates/BaseModel.jinja2 [hunk 4]
```diff
@@ -38,6 +39,13 @@ class {{ class_name }}({{ base_class }}):{% if comment is defined %}  # {{ comme
     {{ method }}
 {%- endfor -%}
 {%- endfor -%}
+{%- for field in extra_fields %}
```
Comment (jezekra1) [inline scripts/types_generator/assets/codegen/templates/BaseModel.jinja2:42]: are `extra_fields` used anywhere?

File: src/genai/schema/_api.py [hunk 5]
```diff
@@ -142,603 +232,472 @@ class ModelType(str, Enum):
...
+        # TODO: remove in next major release
+        deprecated_attrs = {
+            name: getattr(self, name) for name in ["threshold", "send_tokens"] if getattr(self, name, None) is not None
+        }
+        for attr in deprecated_attrs:
+            _log_deprecation_warning(
+                f"genai.schema.ModerationHAP.{attr}",
+                f"Property '{attr}' is deprecated! Use 'input' and 'output' property.",
+            )
+            self.input = self.input or ModerationHAPInput(enabled=True, **deprecated_attrs)
```
Comment (jezekra1) [inline src/genai/schema/_api.py:248]: does this actually work if I have
`ModerationHap(input=True, threshold=0.75)`

The field_validator for `input` will convert it to
`ModerationHAPInput(enabled=True)` so `input` on this line is truthy and the `deprecated_attrs` are not passed or am I missing something?

File: src/genai/extensions/langchain/utils.py [hunk 6]
```diff
@@ -111,3 +112,21 @@ def __add__(self, other: GenerationChunk) -> CustomGenerationChunk:
             )
         else:
             raise TypeError(f"unsupported operand type(s) for +: '{type(self)}' and '{type(other)}'")
+
+
+class CustomAIMessageChunk(AIMessageChunk):
+    """Replaces LangChain's 'merge_dicts' with our simplified 'merge_objects' utility"""
```
Comment (jezekra1) [inline src/genai/extensions/langchain/utils.py:118]: this was a bug? (just curious)

File: src/genai/text/moderation/moderation_service.py [hunk 7]
```diff
@@ -76,6 +77,11 @@ def create(
             ApiNetworkException: In case of unhandled network error.
             ValidationError: In case of provided parameters are invalid.
         """
+        if implicit_hate is not None:
+            self._log_deprecation_warning("'implicit_hate' is deprecated, use 'social_bias' instead!")
```
Comment (jezekra1) [inline src/genai/text/moderation/moderation_service.py:81]: So we ignore them here, but not in generation service?

**Review by jezekra1 — CHANGES_REQUESTED** [review]

File: scripts/types_generator/assets/codegen/templates/BaseModel.jinja2 (hunk 4 above)
Comment (Tomas2D) [reply scripts/types_generator/assets/codegen/templates/BaseModel.jinja2]: Not anymore. Removed.

File: src/genai/schema/_api.py (hunk 5 above)
Comment (Tomas2D) [reply src/genai/schema/_api.py]: It does not; I've oversimplified it. Could you check the implementation now?

File: src/genai/extensions/langchain/utils.py (hunk 6 above)
Comment (Tomas2D) [reply src/genai/extensions/langchain/utils.py:118]: It depends. Their 'merge_dicts' function throws when dictionaries contain an attribute with the `DateTime` data type (in our case, `created_at`)—it does not replace the attribute with the newer one as we do.

File: src/genai/text/moderation/moderation_service.py (hunk 7 above)
Comment (Tomas2D) [reply src/genai/text/moderation/moderation_service.py]: Updated.

File: src/genai/schema/_api.py (hunk 5 above)
Comment (jezekra1) [reply src/genai/schema/_api.py]:
```python
    def model_post_init(self, __context: Any) -> None:
        deprecated_attrs = {
            name: getattr(self, name) for name in ["threshold", "send_tokens"] if getattr(self, name, None) is not None
        }
        for name, value in deprecated_attrs.items():
            _print_deprecation_warning(
                f"(ModerationStigma): '{name}' is deprecated! Use 'input' parameter (ModerationStigmaInput) 'output' parameter (ModerationStigmaOutput) instead.",
            )
            self.input = self.input or ModerationStigmaInput(enabled=True)
            setattr(self.input, name, value)

            self.output = self.output or ModerationStigmaOutput(enabled=True)
            setattr(self.output, name, value)

            delattr(self, name)

    @field_validator("input", mode="before")
    @classmethod
    def _validate_input(cls, value: Any):
        return ModerationStigmaInput(enabled=value) if isinstance(value, bool) else value
```

Won't this enable input if its `None`?
If I have 
`ModerationStigma(output=True, threshold=0.75)`,
looks like this line will enable input even if it was disabled:
`self.input = None or ModerationStigmaInput(enabled=True)`

**Review by jezekra1 — APPROVED** [review]

**Pull request merged**